src/ameba/rule/style/variable_names.cr

```diff
@@ -35,6 +35,18 @@ module Ameba::Rule::Style
       issue_for node, MSG % {expected, node.name}
     end
 
+    # TODO: Handle special case where instance variable is method parameter.
```

Comment:
i think there is no proper way to handle this without changing crystal internals. This was discussed and tried several times

Reply:
I'm thinking create a NodeVisitor subclass that keeps track of the `location` of variables and skips each method parameter that is at the same `location` as an instance variable.

Reply:
Oh, sorry, I confused this with a different issue.